solidity/test/GovernanceTest.js

```diff
@@ -180,50 +180,91 @@ describe("TBTCSystem governance", async function() {
+      await increaseTime(finalizationTime.subn(1))
+
       // Should not revert.
-      await tbtcSystem.beginKeepFactorySingleShotUpdate(
+      await tbtcSystem.beginKeepFactoryUpdate(
+        "0x0000000000000000000000000000000000000003",
+        "0x0000000000000000000000000000000000000004",
+      )
+
+      expect(await tbtcSystem.getRemainingKeepFactoryUpdateTime()).to.eq.BN(
+        await tbtcSystem.getGovernanceTimeDelay(),
+      )
+    })
+
+    it("does not revert if finalizeKeepFactoryUpdate has already been called", async () => {
+      await tbtcSystem.beginKeepFactoryUpdate(
+        "0x0000000000000000000000000000000000000001",
+        "0x0000000000000000000000000000000000000002",
+      )
+
+      const finalizationTime = await tbtcSystem.getRemainingKeepFactoryUpdateTime()
+      await increaseTime(finalizationTime.addn(1))
+
+      await tbtcSystem.finalizeKeepFactoryUpdate()
+
+      // Should not revert.
+      await tbtcSystem.beginKeepFactoryUpdate(
         "0x0000000000000000000000000000000000000001",
         "0x0000000000000000000000000000000000000002",
       )
     })
 
-    it("reverts if finalizeKeepFactorySingleShotUpdate has already been called", async () => {
-      await tbtcSystem.beginKeepFactorySingleShotUpdate(
+    it("does not revert finalizeKeepFactoryUpdate if upgradeability period has passed", async () => {
+      await tbtcSystem.beginKeepFactoryUpdate(
         "0x0000000000000000000000000000000000000001",
         "0x0000000000000000000000000000000000000002",
       )
 
-      const finalizationTime = await tbtcSystem.getRemainingKeepFactorySingleShotUpdateTime()
-      await increaseTime(finalizationTime.toNumber() + 1) // 10 days
-      await tbtcSystem.finalizeKeepFactorySingleShotUpdate()
+      const upgradeabilityTime = await tbtcSystem.getRemainingKeepFactoriesUpgradeabilityTime()
+      await increaseTime(upgradeabilityTime.addn(1))
+
+      await tbtcSystem.finalizeKeepFactoryUpdate()
+    })
+
+    it("reverts for beginKeepFactoryUpdate if upgradeability period has passed", async () => {
+      await tbtcSystem.beginKeepFactoryUpdate(
+        "0x0000000000000000000000000000000000000001",
+        "0x0000000000000000000000000000000000000002",
+      )
+
+      const finalizationTime = await tbtcSystem.getRemainingKeepFactoryUpdateTime()
+      await increaseTime(finalizationTime.addn(1))
+
+      await tbtcSystem.finalizeKeepFactoryUpdate()
+
+      const upgradeabilityTime = await tbtcSystem.getRemainingKeepFactoriesUpgradeabilityTime()
+      await increaseTime(upgradeabilityTime.addn(1))
 
       await expectRevert(
-        tbtcSystem.beginKeepFactorySingleShotUpdate(
+        tbtcSystem.beginKeepFactoryUpdate(
           "0x0000000000000000000000000000000000000001",
           "0x0000000000000000000000000000000000000002",
         ),
-        "Keep factory data can only be updated once",
+        "beginKeepFactoryUpdate can only be called within upgradeability period",
```

Comment:
Let's be specific in this error message---`within 180 days of initialization`.

Reply:
31137e5312e86deac996523b699db69b66291195

Reply:
31137e5312e86deac996523b699db69b66291195